File: AICodingWeb/Models/CustomerTransaction.cs
using System;
using System.Collections.Generic;

namespace AICodingWeb.Models
{
    public class CustomerTransaction
    {
        public int CustomerTransactionID { get; set; }
        public int CustomerID { get; set; }
        public int TransactionTypeID { get; set; }
        public int? InvoiceID { get; set; }
        public int? PaymentMethodID { get; set; }
        public DateTime TransactionDate { get; set; }
        public decimal AmountExcludingTax { get; set; }
        public decimal TaxAmount { get; set; }
        public decimal TransactionAmount { get; set; }
        public decimal OutstandingBalance { get; set; }
        public DateTime? FinalizationDate { get; set; }
        public bool IsFinalized { get; set; }
        public int LastEditedBy { get; set; }
    }
} 

File: AICodingWeb/aicoding_web/app/models/customer_transaction.py
from datetime import datetime
from app import db
from app.models.base import BaseModel

class CustomerTransaction(db.Model, BaseModel):
    """CustomerTransaction model"""
    
    __tablename__ = 'customer_transactions'
    
    CustomerTransactionID = db.Column(db.Integer, primary_key=True)
    CustomerID = db.Column(db.Integer, db.ForeignKey('customers.CustomerID'), nullable=False)
    TransactionTypeID = db.Column(db.Integer, db.ForeignKey('transaction_types.TransactionTypeID'), nullable=False)
    InvoiceID = db.Column(db.Integer, db.ForeignKey('invoices.InvoiceID'))
    PaymentMethodID = db.Column(db.Integer, db.ForeignKey('payment_methods.PaymentMethodID'))
    TransactionDate = db.Column(db.DateTime, nullable=False)
    AmountExcludingTax = db.Column(db.Numeric(18, 2), nullable=False)
    TaxAmount = db.Column(db.Numeric(18, 2), nullable=False)
    TransactionAmount = db.Column(db.Numeric(18, 2), nullable=False)
    OutstandingBalance = db.Column(db.Numeric(18, 2), nullable=False)
    FinalizationDate = db.Column(db.DateTime)
    IsFinalized = db.Column(db.Boolean, nullable=False)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('people.PersonID'), nullable=False)
    LastEditedWhen = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<CustomerTransaction {self.CustomerTransactionID}>' 